core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -711,9 +715,14 @@ static ClientTransport getTransportFromPickResult(PickResult result, boolean isW
       return new ClientTransport() {
         @Override
         public ClientStream newStream(
-            MethodDescriptor<?, ?> method, Metadata headers, CallOptions callOptions) {
-          return transport.newStream(
-              method, headers, callOptions.withStreamTracerFactory(streamTracerFactory));
+            MethodDescriptor<?, ?> method, Metadata headers, CallOptions callOptions,
+            ClientStreamTracer[] tracers) {
+          ClientStreamTracer streamTracer = streamTracerFactory.newClientStreamTracer(
+              StreamInfo.newBuilder()
+                  .setCallOptions(callOptions)
+                  .build());
+          tracers[tracers.length - 1] = streamTracer;
```

Comment:
Assert `tracers[tracers.length - 1] == NOOP_TRACER` before the assignment.

Reply:
Done.